### PR TITLE
Route compared-pane annotation scroll and retry sidebar thread scroll

### PR DESCRIPTION
**Pull request by zhirongwang**
## Summary
- Follow-up to #4842 (already approved). Merge that first; this branch includes it until then.
- While comparing, an annotation on another version is forwarded to that pane instead of switching this one. A request object is used so selecting the same annotation twice still scrolls.
- Activity Feed retries scroll until the active thread is actually in the feed, so a version switch no longer misses it on the first render.

## Test plan
- [ ] Open a PDF with annotations on the current version and a previous version, then start side-by-side compare
- [ ] Click an annotation on the compared pane: compare stays open, that pane scrolls to the annotation
- [ ] Click a compared-version annotation in the activity sidebar: compared pane follows that version if needed and scrolls to it; sidebar scrolls the thread into view
- [ ] Click a current-version annotation in the sidebar: current pane scrolls, compared pane does not steal the scroll
- [ ] Click the same sidebar annotation twice: it still scrolls into view the second time
- [ ] Versions sidebar still changes the compared version as before

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **New Features**
  * Added cross-pane annotation navigation in side-by-side version comparisons.
  * Selecting an annotation in one version now opens and scrolls to the corresponding annotation in the other pane when applicable.

* **Bug Fixes**
  * Prevented annotation-driven version changes from disrupting comparison state.
  * Improved activity feed scrolling while items are still loading.
  * Preserved the correct annotation display settings in compared previews.

### CODE REVIEW

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]
## Verdict: request changes

Follow-up to #4842 looks directionally right: route compare-mode annotation selects by version (`onComparedAnnotationSelect` + `annotationScrollRequest`), keep host `onVersionChange` suppression for annotation returns to current, and retry Activity Feed scroll until the active thread is actually present. Coverage for the new scroll matrix and the feed retry is solid.

Two items before merge:

### 1. PR title fails semantic lint (CI)

`lint_pull_request` failed: title `Route compared-pane annotation scroll and retry sidebar thread scroll` has no conventional-commit type. Rename to something like:

`fix(content-preview): route compared-pane annotation scroll and retry sidebar thread scroll`

### 2. Compare-mode same-pane scroll drops `deferScrollToOnload`

In `handleAnnotationSelect`, the `isComparing` branch for the pane’s own version always calls `emitScrollToAnnotation` and returns, ignoring `deferScrollToOnload` (and the frame/`dynamicOnPreviewLoadAction` path used below). The compared-pane forward path does pass `deferScrollToOnload` through `annotationScrollRequest`, so the two panes are inconsistent. If the current pane’s viewer is not ready yet (common right after opening compare), a same-version sidebar/annotator select can no-op. Please reuse the same defer / on-load scroll path as the non-comparing branch for the same-version comparing case.

### Notes (non-blocking)

- Stacked on #4842 — merge that first as the description says; this branch already includes it.
- `ActivityFeed` re-arming on any `feedItems` reference change is reasonable for version switches given `undefined` means loading; just watch that parents don’t recreate the array every render while an entry is active (would keep re-scrolling).
- Same PREVIEW-1818 nit as on #4842: confirm `showAnnotationsDrawing={false}` doesn’t hide existing drawings on the compared version (create-only disable is the safer read of the docs).

Once the title is fixed and the comparing same-pane path honors defer-on-load, this should be ready to approve.

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]
## Verdict: Request changes

Follow-up to #4842 looks directionally right (route other-version annotation scrolls to the compared pane via a fresh request object; only suppress annotation-origin `onVersionChange` when returning to current while comparing). A few issues should be fixed before merge.

### 1. ActivityFeed re-arms scroll on every `feedItems` identity change (regression)

```147:152:src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js
        if (didLoadFeedItems || hasActiveFeedEntryIdChanged || prevFeedItems !== currFeedItems) {
            this.hasPendingActiveScroll = true;
        }

        if (this.hasPendingActiveScroll) {
            this.scrollToActiveFeedItemOrErrorMessage();
        }
```

After a successful scroll, `hasPendingActiveScroll` is cleared — but the next parent render that passes a new `feedItems` array reference (polling, reply refresh, realtime) sets it again and calls `scrollIntoView` again. That will yank the feed back to the active thread while the user is reading.

**Fix:** Only set `hasPendingActiveScroll = true` on `didLoadFeedItems || hasActiveFeedEntryIdChanged`. Keep calling `scrollToActiveFeedItemOrErrorMessage()` when pending *and* `feedItems` changed (so a later feed update can complete a pending scroll), but do not re-arm after a successful scroll.

### 2. Stale feed on version switch still hits the “missing item” error path

The new `hasActiveFeedItem()` guard only stays pending when the item is **already in** the current feed (or `feedItems === undefined`). On annotation version switches the common sequence is: `activeFeedEntryId` updates first while `feedItems` is still the previous version’s list (item absent → ref null) → `resetFeedScroll()` runs and clears pending → feed jumps to the bottom → later feed update re-arms and scrolls correctly.

That intermediate bottom-scroll fights the stated goal (“version switch no longer misses it on the first render”) and flashes the error path.

**Fix:** When the active id is absent from the current feed, stay pending until `feedItems` itself updates (then decide success vs true-missing). Only `resetFeedScroll()` after a feed update where the id is still missing — not on the id-change tick alone with a stale list. Please add a test for: id changes against a non-empty stale feed that does not contain the id → no `resetFeedScroll`; then new feed without the id → bottom scroll; or new feed with the id + ref → `scrollIntoView`.

### 3. Compared-pane `annotationScrollRequest` has no version guard / mount handling

`annotationScrollRequest` is only consumed in `componentDidUpdate`. When the host must change `comparedVersion` for an annotation on a different older version:

1. The still-mounted compared pane can receive the request first (`isComparing={false}`), so `handleAnnotationSelect` runs the normal “other version” path (`startAt` / version switch) on a pane that is supposed to be pinned to `previewVersion`.
2. After remount (`key={comparedVersionId}`), the same request is an **initial** prop, so `componentDidUpdate` never sees a change; scroll then depends entirely on `fileOptions` / `activeAnnotationId` deep link.

Same-version re-select is covered well; version-follow + scroll is not.

**Fix (either):**
- Only `setAnnotationScrollRequest` when `annotation.file_version.id === comparedVersionId`, and also handle the request in `componentDidMount` (or re-set the request after `comparedVersion` changes so didUpdate fires), **or**
- In the consumer, ignore requests whose `file_version.id` ≠ this pane’s `getVersionToPreview()` id (and ensure remount still scrolls via deep link or a remount-safe request).

A test that simulates “comparing A → select annotation on B” (request + version change) would lock this in.

### 4. PR title fails semantic lint (blocking CI)

`lint_pull_request` failed: title `"Route compared-pane annotation scroll and retry sidebar thread scroll"` has no conventional-commit type. Please rename to something like `fix(content-preview): route compared-pane annotation scroll and retry sidebar thread scroll` (matches the commit / #4842 style).

---

**Nits (non-blocking):** While comparing, same-version path always `emitScrollToAnnotation` and skips the frame/`deferScrollToOnload` branch used off-compare — fine for PDF pages; worth a quick check for video frame annotations on the current pane during compare.

**CI note:** `lint_pull_request` already red on the title; `lint_test_build` was still in progress at review time.

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js [hunk 1]
```diff
@@ -140,27 +142,52 @@ class ActivityFeed extends React.Component<ActivityFeedProps, State> {
             this.resetFeedScroll();
         }
 
-        if (didLoadFeedItems || hasActiveFeedEntryIdChanged) {
+        // Switching file versions replaces the items, so the active entry can be missing from
+        // the feed at the moment its id changes. Re-arm on any feed change to catch it later.
+        if (didLoadFeedItems || hasActiveFeedEntryIdChanged || prevFeedItems !== currFeedItems) {
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js:147]: **Bug / regression:** `prevFeedItems !== currFeedItems` unconditionally re-arms `hasPendingActiveScroll`.

After a successful scroll, any new `feedItems` array identity (parent re-render, pagination, polling, reply refresh) will set the flag again and call `scrollIntoView` — fighting the user’s own scroll position while an active entry stays selected.

That inequality is load-bearing today because the “missing item” branch clears pending when the active id changes before the version’s feed has arrived (old feed still loaded → `hasActiveFeedItem()` false → `resetFeedScroll` + clear). The next feed identity change is what re-arms and eventually scrolls.

Please tighten both sides together, e.g.:
1. Only *set* `hasPendingActiveScroll` on `didLoadFeedItems` / `hasActiveFeedEntryIdChanged` (not on every feed identity change).
2. While already pending, retry on feed updates.
3. Do not treat “active id not in current `feedItems`” as terminal until the feed has actually reloaded for that selection (e.g. only run the missing/error fallback on `didLoadFeedItems`, or stay pending across the first post-id-change feed that still lacks the item).

Also worth a test that a feedItems reference change *after* a successful scroll does **not** call `scrollIntoView` again.

File: src/elements/content-preview/ContentPreview.js [hunk 2]
```diff
@@ -1615,20 +1639,30 @@ class ContentPreview extends React.PureComponent<Props, State> {
...
+        const isOtherVersion = !!annotationFileVersionId && annotationFileVersionId !== currentPreviewFileVersionId;
 
-        if (unit && annotationFileVersionId && annotationFileVersionId !== currentPreviewFileVersionId) {
+        // Each pane stays on its own version for the whole comparison, so route by version
+        // instead of switching: annotations on the compared version belong to the other pane.
+        if (isComparing) {
+            if (isOtherVersion) {
+                onComparedAnnotationSelect(annotation, deferScrollToOnload);
+            } else {
+                this.emitScrollToAnnotation(id, target);
```
Comment (greg-in-a-box) [inline src/elements/content-preview/ContentPreview.js:1660]: **Nit / incomplete path (non-blocking if compare+video is out of scope):** while `isComparing` and the annotation is on this pane’s version, this always `emitScrollToAnnotation` and returns — skipping `deferScrollToOnload` and the frame/video `scrollToFrameAnnotation` wait used on the non-compare path below.

Previously compare bailed out entirely, so this is still an improvement for page annotations, but deferred/video scrolls can still miss. Prefer sharing the same post-routing scroll helper as the non-compare branch (or call into that logic after the version routing decision).

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]
Follow-up to #4842 looks directionally right — compared-pane routing via `onComparedAnnotationSelect` + request-object identity, `origin: 'annotation'` suppression while comparing, and ActivityFeed waiting for the active thread all match the stated goals. Tests cover the main compare/version-origin paths well.

Please address the items below before merge (CI title lint is already red).
